pkg/alertmanager/multitenant.go

```diff
@@ -364,18 +380,36 @@ func (am *MultitenantAlertmanager) addNewConfigs(cfgs map[string]configs.View) {
 // setConfig applies the given configuration to the alertmanager for `userID`,
 // creating an alertmanager if it doesn't already exist.
 func (am *MultitenantAlertmanager) setConfig(userID string, config configs.Config) error {
+	_, hasExisting := am.alertmanagers[userID]
 	amConfig, err := configs_client.AlertmanagerConfigFromConfig(config)
 	if err != nil {
-		// XXX: This means that if a user has a working configuration and
-		// they submit a broken one, we'll keep processing the last known
-		// working configuration, and they'll never know.
-		// TODO: Provide a way of communicating this to the user and for removing
-		// Alertmanager instances.
-		return fmt.Errorf("invalid Cortex configuration for %v: %v", userID, err)
+		if hasExisting {
+			// XXX: This means that if a user has a working configuration and
+			// they submit a broken one, we'll keep processing the last known
+			// working configuration, and they'll never know.
+			// TODO: Provide a way of communicating this to the user and for removing
+			// Alertmanager instances.
+			return fmt.Errorf("invalid Cortex configuration for %v: %v", userID, err)
+		}
+		if am.fallbackConfig != nil {
+			log.Infof("invalid Cortex configuration; using fallback for %v", userID)
+			amConfig = am.fallbackConfig
+		}
+	}
+
+	// Magic ability to configure a Slack receiver if config requests it
+	if am.cfg.FallbackSlackRoot != "" {
+		for _, r := range amConfig.Receivers {
+			for _, s := range r.SlackConfigs {
+				if s.APIURL == "internal://monitor" {
+					s.APIURL = amconfig.Secret(am.cfg.FallbackSlackRoot + "/" + userID + "/monitor")
+				}
+			}
+		}
```

Comment:
This function is getting big enough that I'd consider extracting out the bit that turns `configs.Config` into whatever `amConfig` is. Up to you though.